src/o365/graph/commands/teams/teams-clone.ts

```diff
@@ -0,0 +1,208 @@
+        return 'Required option displayName missing';
+      }
+
+      if (!args.options.mailNickname) {
+        return 'Required option mailNickname missing';
+      }
+
+      if (!args.options.partsToClone) {
+        return 'Required option partsToClone missing';
+      }
+
+      if (args.options.partsToClone) {
+        let partsToClone: string[] = args.options.partsToClone.split(',');
+
+        for (let partToClone of partsToClone) {
+
+          let part: string = partToClone.toLowerCase();
+
+          if (part !== 'apps' &&
+            part !== 'channels' &&
+            part !== 'members' &&
+            part !== 'settings' &&
+            part !== 'tabs') {
+            return `${part} is not a valid partsToClone. Allowed values are apps|channels|members|settings|tabs`;
+          }
+        }
+      }
+
+      if (args.options.visibility) {
+        const visibility: string = args.options.visibility.toLowerCase();
+
+        if (visibility !== 'private' &&
+          visibility !== 'public') {
+          return `${args.options.visibility} is not a valid visibility type. Allowed values are Private|Public`;
+        }
+      }
+
+      return true;
+    };
+  }
+
+  public commandHelp(args: {}, log: (help: string) => void): void {
+    const chalk = vorpal.chalk;
+    log(vorpal.find(this.name).helpInformation());
+    log(
+      `  ${chalk.yellow('Important:')} Before using this command, log in to the Microsoft Graph,
+    using the ${chalk.blue(commands.LOGIN)} command.
+          
+  Remarks:
+          
+    To clone a Microsoft Teams team, you have to first log in to the Microsoft
+    Graph using the ${chalk.blue(commands.LOGIN)} command, eg. ${chalk.grey(`${config.delimiter} ${commands.LOGIN}`)}.
+
+    Using this command, global admins and Microsoft Teams service admins can
+    access teams that they are not a member of.
+
+    When tabs are cloned, they are put into an unconfigured state and they are 
+    displayed on the tab bar in Microsoft Teams, and the first time you open them, 
+    you'll go through the configuration screen. (If the person opening the tab does 
+    not have permission to configure apps, they will see a message explaining that 
+    the tab hasn't been configured.)
+
+  Examples:
+    
+    Creates a clone of a Microsoft Teams team with mandatory parameters
+      ${chalk.grey(config.delimiter)} ${commands.TEAMS_CLONE} --teamId 15d7a78e-fd77-4599-97a5-dbb6372846c5 --displayName "Library Assist" --mailNickname "libassist" --partsToClone "apps,tabs,settings,channels,members" 
+    
+    Creates a clone of a Microsoft Teams team with mandatory and optional parameters
+      ${chalk.grey(config.delimiter)} ${commands.TEAMS_CLONE} --teamId 15d7a78e-fd77-4599-97a5-dbb6372846c5 --displayName "Library Assist" --mailNickname "libassist" --partsToClone "apps,tabs,settings,channels,members" --description "Self help community for library" --classification "Library" --visibility "public"
```

Comment:
To drop `--mailNickname "libassist" ` from the example. 
See: https://github.com/pnp/office365-cli/pull/939#discussion_r279938818

Reply:
@VelinGeorgiev, This is complete.